service/src/main/java/bio/terra/externalcreds/dataAccess/NihAccountDAO.java

```diff
@@ -0,0 +1,108 @@
+package bio.terra.externalcreds.dataAccess;
+
+import bio.terra.externalcreds.models.NihAccount;
+import io.opentelemetry.instrumentation.annotations.WithSpan;
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.dao.support.DataAccessUtils;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.stereotype.Repository;
+
+@Repository
+@Slf4j
+public class NihAccountDAO {
+
+  private static final RowMapper<NihAccount> NIH_ACCOUNT_ROW_MAPPER =
+      ((rs, rowNum) ->
+          new NihAccount.Builder()
+              .id(rs.getInt("id"))
+              .userId(rs.getString("user_id"))
+              .nihUsername(rs.getString("nih_username"))
+              .expires(rs.getTimestamp("expires"))
+              .build());
+
+  final NamedParameterJdbcTemplate jdbcTemplate;
+
+  public NihAccountDAO(NamedParameterJdbcTemplate jdbcTemplate) {
+    this.jdbcTemplate = jdbcTemplate;
+  }
+
+  @WithSpan
+  public Optional<NihAccount> getNihAccount(String userId) {
+    var namedParameters = new MapSqlParameterSource().addValue("userId", userId);
+    var query = "SELECT * FROM nih_account WHERE user_id = :userId";
+    return Optional.ofNullable(
+        DataAccessUtils.singleResult(
+            jdbcTemplate.query(query, namedParameters, NIH_ACCOUNT_ROW_MAPPER)));
+  }
+
+  @WithSpan
+  public Optional<NihAccount> getNihAccountForUsername(String nihUsername) {
+    var namedParameters = new MapSqlParameterSource().addValue("nihUsername", nihUsername);
+    var query = "SELECT * FROM nih_account WHERE nih_username = :nihUsername";
+    return Optional.ofNullable(
+        DataAccessUtils.singleResult(
+            jdbcTemplate.query(query, namedParameters, NIH_ACCOUNT_ROW_MAPPER)));
+  }
+
+  @WithSpan
+  public List<NihAccount> getExpiredNihAccounts() {
+    var namedParameters =
+        new MapSqlParameterSource("expirationCutoff", Timestamp.from(Instant.now()));
+    var query =
+        "SELECT DISTINCT na.* FROM nih_account na" + " WHERE na.expires < :expirationCutoff";
```

Comment:
distinct is sort of a smell imo, is there a way to do this with a group by? also listing out the columns explicitly would be nice

Reply:
I don't actually think we need to `distinct` or `group by` here at all. The unique constrains on the table should take care of that.